### PR TITLE
Clickable datapoints

### PR DESCRIPTION
**Pull request by Velenir**
Adds clickable datapoints to Orderbook chart
To set price from there

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1316--dexreact.review.gnosisdev.com

File: src/components/TradeWidget/index.tsx [hunk 1]
```diff
@@ -488,6 +489,20 @@ function buildUrl(params: {
...
+}
+
+const OrderbookContext = React.createContext<OrderbookContextI>({
+  setSellToken: () => void 0,
+  setReceiveToken: () => void 0,
+  setPrice: () => void 0,
+})
+
+export const useOrderbookContext = (): OrderbookContextI => React.useContext(OrderbookContext)
+
```
Comment (alfetopito) [inline src/components/TradeWidget/index.tsx:505]: nit: move to a hook file?

**Comment by Velenir** [comment]
> Shouldn't the volume be filled in as well?
Do we want that? A user may want to sell specific amount and went to orderbook to look up a good price

> whatever is currently displayed in the tooltip be set. Can that be done?

I wonder. Tooltip isn't clickable by itself, it seems. And just clicking anywhere gets the whole chart as an `env.target`. Therefore dataPoints is the only specific point that is directly clickable. StepLineSeries uses those same dataPoints to draw steps, and to sticky tooltips to. Just as we can't put a datapoint in the middle of a step, can make it a  clickable target.

And if we just detect what is in the open tooltip when we click anywhere it will lead to misclicks as tooltips snap to steps, and cursor can be rather far off

**Comment by Velenir** [comment]
We should discuss all this separately.
Therefore this is a draft to see what is possible

> if the user clicks in a point in the graph, one idea to iterate is, he sets a horizontal and vertical line

And then the user would like to move those lines, and that's a whole lot of extra code

> I don't think you are handling the case that the user changes the tokens in the current modal

Handling already, by changing URl and tokens

**Pull request closed without merging**